app/models/changeset/pull_request.rb

```diff
@@ -38,9 +44,19 @@ def self.changeset_from_webhook(project, params = {})
     new(project.github_repo, data)
   end
 
+  # Webhook events that are valid should be related to a pr code push or someone adding [samson review]
+  # to the description. The actions related to a code push are 'opened' and 'synchronized'
+  # The 'edited' action gets sent when the PR description is edited. To trigger a deploy from an edit - it
+  # should only be when the edit is related to adding the text [samson review]
   def self.valid_webhook?(params)
     data = params['pull_request'] || {}
-    return false unless data['state'] == 'open'
+    action = params.fetch('github').fetch('action')
```

Comment:
we just moved to ruby 2.3, so you can use [this](http://ruby-doc.org/core-2.3.0_preview1/Hash.html#method-i-dig) if you like